hlc: move ParseHLC / DecimalToHLC to util/hlc

otan opened the pull request:
util/hlc needs util/log, but ParseHLC/DecimalToHLC doesn't need to be
in tree. To remove the tree dependency on util/log, move the
mentioned functions.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/80892)
<!-- Reviewable:end -->

otan committed:
hlc: move ParseHLC / DecimalToHLC to util/hlc

util/hlc needs util/log, but ParseHLC/DecimalToHLC doesn't need to be
in tree. To remove the tree dependency on util/log, move the
mentioned functions.

Release note: None

ajwerner reviewed (approved):
> Reviewed 22 of 22 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/80892)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @otan)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=ajwerner
